src/app/components/utility/input/input.component.html

```diff
@@ -1,11 +1,20 @@
 <div class="input-group">
-  <div *ngIf="type!='file'">    
-  <input class="input-field" (input)="validateInput($event.target.value)" [class.isValid]="isValid" [class.theme-dark]="theme=='dark'" type="{{type}}" [value]="value" [readonly]="isReadonly">
-  <span class="input-bar"></span>
-  <label [class.is-not-empty]="!isEmpty" [class.theme-dark]="theme=='dark'">{{placeholder}}</label>
-  <img class="input-icon" src="{{icon}}" *ngIf="isIconPresent" />
-  <div class="input-message" #textValue [class.hidden]="!(((isRequired && isEmpty)||(!isValid && !isEmpty))  && isDirty)">{{message}}</div>
+  <div *ngIf="type!='file' && type!='textarea'">
+    <input class="input-field" (input)="validateInput($event.target.value)" [class.isValid]="isValid" [class.theme-dark]="theme=='dark'" type="{{type}}" [value]="value" [readonly]="isReadonly">
+    <span class="input-bar"></span>
+    <label [class.is-not-empty]="!isEmpty" [class.theme-dark]="theme=='dark'">{{placeholder}}</label>
+    <i class="input-icon {{icon}}" *ngIf="isIconPresent"></i>
+    <div class="input-message" #textValue [class.hidden]="!(((isRequired && isEmpty)||(!isValid && !isEmpty))  && isDirty)">{{message}}</div>
```

Comment:
conditions like `!(((isRequired && isEmpty)||(!isValid && !isEmpty))  && isDirty)` is not looking good and readable. You can create a separate method to check it- which will return True/False.

Reply:
@Shekharrajak done